File: C/ActionLoggerc/ActionLoggerc/ActionLogger.c
#include <stdio.h>
#include <sys/stat.h>

void log(const char* s)
{
    FILE *outfile;
    struct stat st;
    if (stat("action_logs.txt", &st) == 0) {
        outfile = fopen("action_logs.txt", "a");
        fprintf(outfile, "\n%s", s);
    }
    else {
        outfile = fopen("action_logs.txt","w");
        fprintf(outfile, "%s", s);
    }
    fclose(outfile);

}

main()
{
    log("First message");
    log("Second message");
}

File: Python/ActionLogger.py
import pathlib
def log(msg):
    if pathlib.Path("action_logs.txt").exists:
        with open("action_logs.txt", "a") as file:
            file.write("\n"+msg)
    else:
        with open("action_logs.txt", "w") as file:
            file.write(msg)
log("first message")
log("second message")
